### PR TITLE
feat(java): use java 11 in favour of java 8.

### PR DESCRIPTION
**Pull request by walschap**
When running gradle using java 8 on a project that depends on libraries built with newer java versions, as of gradle 5.3 the renovate run breaks because of the "Automatic target JVM version" check. -->  https://docs.gradle.org/current/userguide/upgrading_version_5.html#changes_5.3

Closes #

### CODE REVIEW

**Comment by rarkins** [comment]
Hi @walschap, are you using the self-hosted Renovate, or WhiteSource Renovate app? We need to update https://github.com/renovatebot/docker-gradle/blob/master/Dockerfile too

**Comment by ViceIce** [comment]
Currently migrating to java11 would break android gradle projects, because there are some missing libs removed from java to external libs. 🤔

**Comment by walschap** [comment]
Yes @rarkins , we're using the self-hosted renovate.

**Comment by ViceIce** [comment]
@rarkins android already broken on app

https://github.com/ViceIce/android-samples
https://app.renovatebot.com/dashboard#github/ViceIce/android-samples/130730044

**Comment by ViceIce** [comment]
strange, on my selfhosted `renovate:slim` i get the follwing:

```
DEBUG: Start gradle command (repository=proj/repo)
       "cmd": "docker run --rm -v \"/tmp/renovate/repos/gitlab/proj/repo\":\"/tmp/renovate/repos/gitlab/proj/repo\" -w \"/tmp/renovate/repos/gitlab/proj/repo\" renovate/gradle gradle --init-script renovate-plugin.gradle renovate"
DEBUG: To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/5.4.1/userguide/gradle_daemon.html.
       Daemon will be stopped at the end of the build stopping after processing
       Configuration on demand is an incubating feature.
       > Task :app:renovate
       > Task :renovate
       
       BUILD SUCCESSFUL in 38s
       2 actionable tasks: 2 executed
        (repository=proj/repo)
 INFO: Gradle report complete (sitory=proj/repo)
```

**Comment by rarkins** [comment]
@ViceIce any objections to moving to Java 11 for the fat Docker image here then?

**Comment by ViceIce** [comment]
I'm just testing it now, maybe we can upgrade to java11 and update android projects without android sdk

**Comment by ViceIce** [comment]
ok, i got it running. https://github.com/ViceIce/android-samples/pulls

I will try to run with java 11 now

**Comment by ViceIce** [comment]
it's working fine, even with java 13

**Comment by ViceIce** [comment]
@rarkins can we merge as feature, or does it need a breaking change 🤔 

Also we need to update `renovate/gradle` docker image. I'll send a pr

**Comment by ViceIce** [comment]
`renovate/gradle` is already on java11

https://github.com/renovatebot/docker-java/blob/b0c3e3bcac1c8de7f7a1a9767df70c7a6c841892/Dockerfile#L5

we install [`default-jre-headless`](https://packages.ubuntu.com/bionic/default-jre-headless) which will  install java11, so it should be save to update the full image without breaking change.

File: Dockerfile [hunk 1]
```diff
@@ -29,7 +29,7 @@ RUN apt-get update && apt-get install -y gpg curl wget unzip xz-utils git openss
 
 ## Gradle
 
-RUN apt-get update && apt-get install -y --no-install-recommends openjdk-8-jre-headless gradle && \
+RUN apt-get update && apt-get install -y --no-install-recommends openjdk-11-jre-headless gradle && \
```
Comment (ViceIce) [inline Dockerfile]:
```suggestion
RUN apt-get update && apt-get install -y --no-install-recommends gradle && \
```

gradle installs latest stable java version (`default-jre-headless`)

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by ViceIce** [comment]
image

**Pull request merged**